src/test/java/com/rocketden/main/UserIntegrationTests.java

```diff
@@ -96,7 +98,62 @@ public void handleFrame(StompHeaders headers, Object payload) {
                     }
                 });
                 try {
-                    session.send(BaseRestController.BASE_SOCKET_URL + "/user", "Chris");
+                    session.send(BaseRestController.BASE_SOCKET_URL + "/add-user", "Chris");
+                } catch (Throwable t) {
+                    failure.set(t);
+                    latch.countDown();
+                }
+            }
+        };
+
+        // Connect to the socket with the STOMP client.
+        this.stompClient.connect("ws://localhost:{port}" + BaseRestController.BASE_SOCKET_URL + "/join-room-endpoint", this.headers, handler, this.port);
+
+        if (latch.await(3, TimeUnit.SECONDS)) {
+            if (failure.get() != null) {
+                throw new AssertionError("", failure.get());
+            }
+        } else {
+            fail("User response not received");
+        }
+
+    }
+
+    @Test
+    public void deleteUser() throws Exception {
+
+        // Latch allows threads to wait until remaining operations complete.
+        final CountDownLatch latch = new CountDownLatch(1);
+        final AtomicReference<Throwable> failure = new AtomicReference<>();
+
+        // Test session handler for STOMP (created below).
+        StompSessionHandler handler = new TestSessionHandler(failure) {
+
+            @Override
+            public void afterConnected(final StompSession session, StompHeaders connectedHeaders) {
+                // After the socket is connected, subscribe and send message.
+                session.subscribe(BaseRestController.BASE_SOCKET_URL + "/subscribe-user", new StompFrameHandler() {
+                    @Override
+                    public Type getPayloadType(StompHeaders headers) {
+                        return HashSet.class;
+                    }
+
+                    @Override
+                    public void handleFrame(StompHeaders headers, Object payload) {
+                        Set<User> users = (HashSet<User>) payload;
+                        try {
+                            // Verify that the subscription received the expected message.
+                            assertEquals(0, users.size());
+                        } catch (Throwable t) {
+                            failure.set(t);
+                        } finally {
+                            session.disconnect();
+                            latch.countDown();
+                        }
+                    }
+                });
+                try {
+                    session.send(BaseRestController.BASE_SOCKET_URL + "/delete-user", "Chris");
```

Comment:
Is Chris already in the HashSet when you run this or is this just making sure removing a nonexistent user doesn't cause an error?

Reply:
I'm removing a nonexistent error. I noted in the "Notes" section of the PR description that I had trouble creating two "send" functions on the same `@Test`, so I'm not sure how to test actually removing a user (or any sequence of socket send operations).

Reply:
I added issue #33 for this. I'll merge this PR, but we should address this at some point.